### PR TITLE
[SPARK-37492][SQL] Optimize Orc test code with withAllNativeOrcReaders

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
This PR used to optimize some Orc test code in OrcSourceSuite.scala

### Why are the changes needed?
Improve code.

### Does this PR introduce _any_ user-facing change?
'No'. Just improve code for UT.

### How was this patch tested?
Jenkins test.

### CODE REVIEW

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Sorry, but hold on please, @beliefer .

Although this PR itself is correct, the previous commit seems to misleading.
`withAllOrcReaders` is not correct in Apache Spark context because this only tests `native` OrcReaders. We have `hive` ORC reader too.

**Comment by dongjoon-hyun** [comment]
Let's proceed this after https://github.com/apache/spark/pull/34733 .

**Comment by dongjoon-hyun** [comment]
#34733 is merged. Could you rebase this PR to the master branch, @beliefer ?

**Comment by dongjoon-hyun** [comment]
BTW, is this PR include all instances like this, @beliefer ? Could you check other places too?

**Comment by beliefer** [comment]
> BTW, is this PR include all instances like this, @beliefer ? Could you check other places too?

Yes. I checked every places.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @beliefer and @srowen .
Merged to master.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcQuerySuite.scala [hunk 1]
```diff
@@ -288,10 +288,9 @@ class HiveOrcQuerySuite extends OrcQueryTest with TestHiveSingleton {
 
   test("SPARK-32234 read ORC table with column names all starting with '_col'") {
     Seq("native", "hive").foreach { orcImpl =>
-      Seq("false", "true").foreach { vectorized =>
-        withSQLConf(
-          SQLConf.ORC_IMPLEMENTATION.key -> orcImpl,
-          SQLConf.ORC_VECTORIZED_READER_ENABLED.key -> vectorized) {
+      withSQLConf(
+        SQLConf.ORC_IMPLEMENTATION.key -> orcImpl) {
+        withAllNativeOrcReaders {
```
Comment (dongjoon-hyun) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcQuerySuite.scala:293]: Ur, @beliefer . This is technically correct, but logically misleading again. When `orcImpl=hive`, we are not testing `nativeOrcReaders`.

Comment (dongjoon-hyun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcQuerySuite.scala:293]: I'll merge this PR without this part.

Comment (beliefer) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcQuerySuite.scala:293]: @dongjoon-hyun Thank you for the reminder.

**Pull request closed without merging**